### PR TITLE
fix `ls --json` being corrupted by `Using https://...` msg

### PR DESCRIPTION
**Pull request by mjurbanski-reef**

### CODE REVIEW

File: b2/console_tool.py [hunk 1]
```diff
@@ -1991,30 +1995,33 @@ def _setup_parser(cls, parser):
...
-            #   That would sadly lead to an ugly formatting, so `_print` needs to be expanded with an ability
-            #   to not print end-line character(s).
-            self._print_json([file_version for file_version, _ in self._get_ls_generator(args)])
-            return 0
-
-        return super().run(args)
+            if self._first_row:  # no files were printed
+                self._print_json('[', raw=True)
+            self._print_json(']\n', raw=True)
+        return 0
```
Comment (agoncharov-reef) [inline b2/console_tool.py:2004]: I don't get it. How is it supposed to work?
Why not something stupid like
```python
if args.json:
    self._print('[')
    for i, (file_version, _) in enumerate(self._get_ls_generator(args)):
         if i:
            self._print(', ', print_endline=False)
         self._print_json(file_version)
    self._print(']')
else:
     self._print_files(args)
```

File: test/unit/test_console_tool.py [hunk 2]
```diff
@@ -84,7 +84,9 @@ def _run_command_ignore_output(self, argv):
             print('ACTUAL STDERR:  ', repr(actual_stderr))
             print(actual_stderr)
 
-        self.assertEqual('', actual_stderr, 'stderr')
+        assert re.match(
+            r'^(|Using https?://[\w.]+.com\n)$', actual_stderr
```
[the remark this answers is not in the document]
Comment (mjurbanski-reef) [reply test/unit/test_console_tool.py:88]: (|xxx) catches empty line as well as xxx
and afaik, I do need \n there

File: b2/console_tool.py [hunk 3]
```diff
@@ -685,23 +685,28 @@ def _parse_file_infos(cls, args_info):
             file_infos[parts[0]] = parts[1]
         return file_infos
 
-    def _print_json(self, data):
-        self._print(
-            json.dumps(data, indent=4, sort_keys=True, cls=B2CliJsonEncoder), enforce_output=True
-        )
+    def _print_json(self, data, *, raw: bool = False):
```
[the remark this answers is not in the document]
Comment (mjurbanski-reef) [reply b2/console_tool.py:688]: the reason was that I didn't want to expose `enforce_output` to anything else that method printing actual data - that would invite someone to break intended `--quiet` behavior.

I guess I will document it in docstring instead.

**Comment by mjurbanski-reef** [comment]
squash merged in the upstream

**Pull request closed without merging**